src/snowcli/cli/snowpark/function/commands.py

```diff
@@ -110,32 +118,47 @@ def function_create(
     sm = StageManager()
     fm = FunctionManager()
 
-    artifact_file = upload_snowpark_artifact(
-        function_manager=fm,
-        stage_manager=sm,
-        function_identifier=identifier,
-        file=file,
-        overwrite=overwrite,
-    )
+    result = MultipleResults()
+    if function_names:
+        functions = [f for f in functions if f["name"] in function_names]
 
-    packages = get_snowflake_packages()
-    cursor = fm.create(
-        identifier=identifier,
-        handler=handler,
-        return_type=return_type,
-        artifact_file=str(artifact_file),
-        packages=packages,
-        overwrite=overwrite,
-    )
-    return SingleQueryResult(cursor)
+    if not functions:
+        return MessageResult("No functions defined in this project.")
+
+    sm.create(stage_name=DEPLOYMENT_STAGE, comment="deployments managed by snowcli")
+
+    for function in functions:
+        arguments = ", ".join(
+            (f"{arg['name']} {arg['type']}" for arg in function["signature"])
+        )
+        identifier = f"{function['name']}({arguments})"
```

Comment:
Interesting point, I would say that in case of overloading those functions are the "same" function. Why otherwise use the same name?

Reply:
Yes, they can be treated as the "same" function, but they are not the same function.

An overloaded variant of function is a separate function in snowflake and it can have totally different implementation. So if we allow to deploy a specific function, then it's not very clear why don't we allow to deploy only a specific overloaded version of a function.

For example we allow for separate deployment of the following functions:
```
sum_ints(x: int, y: int) -> int
sum_ints_represented_as_strings(x: string, y: string) -> int
```
But we don't allow for separate deployment of the following functions:
```
sum(x: int, y: int) -> int
sum(x: string, y: string) -> int
```

CLI options approach allowed for separate deployment in both cases. Snowflake supports both cases but our CLI won't do.

I'm not saying I see use cases for the second scenario, but I don't see them also for the first scenario.